File: src/main/java/com/example/jwttest/dto/TokenDto.java
package com.example.jwttest.dto;


/**
 * Response 에 사용할 Token Dto
 */
public class TokenDto {

    private String token;

    private String refreshToken;

    public TokenDto(String token, String refreshToken) {
        this.token = token;
        this.refreshToken = refreshToken;
    }

    public TokenDto() {
    }

    public static TokenDtoBuilder builder() {
        return new TokenDtoBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getRefreshToken() {
        return this.refreshToken;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public static class TokenDtoBuilder {
        private String token;
        private String refreshToken;

        TokenDtoBuilder() {
        }

        public TokenDtoBuilder token(String token) {
            this.token = token;
            return this;
        }

        public TokenDtoBuilder refreshToken(String refreshToken) {
            this.refreshToken = refreshToken;
            return this;
        }

        public TokenDto build() {
            return new TokenDto(token, refreshToken);
        }

        public String toString() {
            return "TokenDto.TokenDtoBuilder(token=" + this.token + ", refreshToken=" + this.refreshToken + ")";
        }
    }
}

File: src/main/kotlin/com/example/jwttest/dto/TokenDto.kt
package com.example.jwttest.dto

/**
 * Response 에 사용할 Token Dto
 */
class TokenDto {
    var token: String? = null
    var refreshToken: String? = null

    constructor(token: String?, refreshToken: String?) {
        this.token = token
        this.refreshToken = refreshToken
    }

    constructor()

    class TokenDtoBuilder internal constructor() {
        private var token: String? = null
        private var refreshToken: String? = null
        fun token(token: String?): TokenDtoBuilder {
            this.token = token
            return this
        }

        fun refreshToken(refreshToken: String?): TokenDtoBuilder {
            this.refreshToken = refreshToken
            return this
        }

        fun build(): TokenDto {
            return TokenDto(token, refreshToken)
        }

        override fun toString(): String {
            return "TokenDto.TokenDtoBuilder(token=" + token + ", refreshToken=" + refreshToken + ")"
        }
    }

    companion object {
        fun builder(): TokenDtoBuilder {
            return TokenDtoBuilder()
        }
    }
}